pom.xml

```diff
@@ -141,6 +141,17 @@
             <forceJavacCompilerUse>true</forceJavacCompilerUse>
             <source>${java.version}</source>
             <target>${java.version}</target>
+            <compilerArgs>
+              <arg>-Xplugin:Traute</arg>
+              <arg>-Atraute.failure.text.parameter=$${PARAMETER_NAME} == null</arg>
+            </compilerArgs>
+            <annotationProcessorPaths>
+              <path>
+                <groupId>tech.harmonysoft</groupId>
+                <artifactId>traute-javac</artifactId>
```

Comment:
Docs seem to suggest this will mean compiling only with JDK8+

https://github.com/denis-zhdanov/traute/blob/master/core/javac/README.md

I'm not certain this is a problem, but certainly could be for any downstream build processes.

Reply:
Yes, *Traute* is a *javac* plugin and corresponding *API* is available only since *java8*.  

There are clients who use something like `javac8 -source 1.7 -target 1.7` and *Traute* works perfectly for them. It doesn't work if you use *javac7* though.

Reply:
We should fix those tests